src/hooks/useCogsEvent.ts

```diff
@@ -1,22 +1,24 @@
-import { CogsConnection, EventKeyValue } from '@clockworkdog/cogs-client';
+import { CogsConnection, CogsIncomingEventTypes, ManifestTypes } from '@clockworkdog/cogs-client';
 import { useEffect } from 'react';
+import { ManifestFromConnection } from '../utils/types';
 
-type InputEventType<Connection> = Connection extends CogsConnection<infer Types> ? NonNullable<Types['inputEvents']> : never;
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export default function useCogsEvent<Connection extends CogsConnection<any>, EventKey extends keyof InputEventType<Connection>>(
+export default function useCogsEvent<
```

Comment:
In my testing of this, I'm trying to extract the `value` from the `event` with an Event type of `option` but the type always seems to be `undefined`